checker/src/org/checkerframework/checker/index/samelen/SameLenTransfer.java

```diff
@@ -73,16 +75,46 @@ public SameLenTransfer(CFAnalysis analysis) {
                                 UNKNOWN,
                                 arrayLengthNodeAnnotation);
 
-                result.getRegularStore().clearValue(targetRec);
-                result.getRegularStore().insertValue(targetRec, combinedSameLen);
-                result.getRegularStore().clearValue(otherRec);
-                result.getRegularStore().insertValue(otherRec, combinedSameLen);
+                propagateCombinedSameLen(combinedSameLen, node, result.getRegularStore());
             }
         }
 
+        // If the left side of the assignment has a SameLen type, propogate the right side to the arrays listed there.
+
+        AnnotationMirror leftAnno =
+                aTypeFactory
+                        .getAnnotatedType(node.getExpression().getTree())
+                        .getAnnotationInHierarchy(UNKNOWN);
+
+        if (leftAnno != null && AnnotationUtils.areSameByClass(leftAnno, SameLen.class)) {
+
+            AnnotationMirror combinedSameLen =
+                    aTypeFactory.createCombinedSameLen(
+                            node.getTarget().toString(),
+                            node.getExpression().toString(),
+                            UNKNOWN,
+                            leftAnno);
+
+            propagateCombinedSameLen(combinedSameLen, node, result.getRegularStore());
+        }
+
         return result;
     }
 
+    private void propagateCombinedSameLen(
+            AnnotationMirror combinedSameLen, Node node, CFStore store) {
+        // Get the value of the combined samelen type, and then
+        // insert that combined SameLen into the receiver for each.
```

Comment:
I think you mean"insert the combined SameLen into the store for each array mentioned in the SameLen".